base/linalg/bunchkaufman.jl

```diff
@@ -104,6 +104,71 @@ function det(F::BunchKaufman)
     return d
 end
 
+function logabsdet(F::BunchKaufman)
+    M = F.LD
+    p = F.ipiv
+    n = size(F.LD, 1)
+
+    s = one(real(eltype(F)))
+    i = 1
+    abs_det = zero(real(eltype(F)))
+    while i <= n
+        if p[i] > 0
+            elm = M[i,i]
+            s *= sign(elm)
+            abs_det += log(abs(elm))
+            i += 1
+        else
+            if F.uplo == 'U'
+                elm = M[i, i + 1]*(M[i,i]/M[i, i + 1]*M[i + 1, i + 1] - (issym(F) ? M[i, i + 1] : conj(M[i, i + 1])))
+                s *= sign(elm)
+                abs_det += log(abs(elm))
+            else
+                elm = M[i + 1,i]*(M[i, i]/M[i + 1, i]*M[i + 1, i + 1] - (issym(F) ? M[i + 1, i] : conj(M[i + 1, i])))
+                s *= sign(elm)
+                abs_det += log(abs(elm))
+            end
+            i += 2
+        end
+    end
+    return abs_det, s
+end
+
+function logdet{T<:Real}(F::BunchKaufman{T})
+    d, s = logabsdet(F)
+    if s < 0
+        throw(DomainError())
+    end
+    d
+end
+
+_mod2pi(x::BigFloat) = mod(x, big(2)*π) # we don't want to export this, but we use it below
+_mod2pi(x) = mod2pi(x)
+function logdet{T<:Complex}(F::BunchKaufman{T})
```

Comment:
I think you have done all the hard work already in `logabsdet` so it should be sufficient with something like
function logdet(F::BunchKaufman)
    d, s = logabsdet(F)
    return d + log(s)
end
```
Notice that this also works in the real case since `log` will throw the right error when `s<0`.